Content.Server/Antag/MobReplacementRuleSystem.cs

```diff
@@ -1,37 +1,172 @@
+using System.Numerics;
 using Content.Server.Antag.Mimic;
+using Content.Server.Chat.Systems;
 using Content.Server.GameTicking.Rules;
 using Content.Server.GameTicking.Rules.Components;
+using Content.Server.NPC.Systems;
+using Content.Server.Station.Systems;
+using Content.Server.GameTicking;
 using Content.Shared.VendingMachines;
 using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
 using Robust.Shared.Random;
+using Robust.Server.GameObjects;
+using Robust.Shared.Physics.Systems;
+using System.Linq;
+using Robust.Shared.Physics;
+using Content.Shared.Movement.Components;
+using Content.Shared.Damage;
+using Content.Server.NPC.HTN;
+using Content.Server.NPC;
+using Content.Shared.Weapons.Melee;
+using Content.Server.Advertise;
+using Content.Server.Power.Components;
+using Content.Shared.CombatMode;
 
 namespace Content.Server.Antag;
 
 public sealed class MobReplacementRuleSystem : GameRuleSystem<MobReplacementRuleComponent>
 {
     [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+    [Dependency] private readonly GameTicker _gameTicker = default!;
+    [Dependency] private readonly ChatSystem _chatSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly IComponentFactory _componentFactory = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] private readonly NpcFactionSystem _npcFaction = default!;
+    [Dependency] private readonly NPCSystem _npc = default!;
+    [Dependency] private readonly TransformSystem _transform = default!;
+    [Dependency] private readonly AdvertiseSystem _advertise = default!;
 
     protected override void Started(EntityUid uid, MobReplacementRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
         base.Started(uid, component, gameRule, args);
 
         var query = AllEntityQuery<VendingMachineComponent, TransformComponent>();
         var spawns = new List<(EntityUid Entity, EntityCoordinates Coordinates)>();
+        var stations = _gameTicker.GetSpawnableStations();
 
         while (query.MoveNext(out var vendingUid, out _, out var xform))
         {
-            if (!_random.Prob(component.Chance))
+            var ownerStation = _stationSystem.GetOwningStation(vendingUid);
+
+            if (ownerStation == null
+                || ownerStation != stations[0])
+                return;
```

Comment:
Shouldn't this be a continue?

```suggestion
                continue;
```